src/history/HistoryArchiveReportWork.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "history/HistoryArchiveReportWork.h"
+#include "historywork/GetHistoryArchiveStateWork.h"
+#include "util/Logging.h"
+#include <Tracy.hpp>
+#include <fmt/format.h>
+
+namespace stellar
+{
+HistoryArchiveReportWork::HistoryArchiveReportWork(
+    Application& app,
+    std::vector<std::shared_ptr<HistoryArchive>> const& archives)
+    : Work(app, "history-archive-report-work", BasicWork::RETRY_NEVER)
+    , mGetHistoryArchiveStateWorks()
+    , mAdded(false)
+    , mArchives(archives)
+{
+}
+
+BasicWork::State
+HistoryArchiveReportWork::doWork()
+{
+    ZoneScoped;
+
+    if (!mAdded)
+    {
+        for (auto const& archive : mArchives)
+        {
+            std::shared_ptr<GetHistoryArchiveStateWork> p =
+                addWork<GetHistoryArchiveStateWork>(
+                    0, archive, "archive-report-" + archive->getName(),
+                    BasicWork::RETRY_NEVER);
+            mGetHistoryArchiveStateWorks.push_back(p);
+            mChildren.push_back(p);
+        }
+        mAdded = true;
+    }
+
+    if (allChildrenDone())
+    {
+        // If some work fails quickly and gets cleaned up,
+        // then allChildrenSuccessful() returns true despite the failed child.
+        // Thus we want to iterate through all the children in order to return
+        // the correct state.
+        if (WorkUtils::allSuccessful(mChildren))
```

Comment:
parent work doesn't get scheduled that often (mostly it's just waiting for the downloads), so I don't think you need `mChildren`. Just create the list on the go, like we do it here: https://github.com/stellar/stellar-core/blob/c097a3fe36f108bcbb52e4995b81bfe617f3109a/src/historywork/PutSnapshotFilesWork.cpp#L70

Reply:
Got it! Fixed.